lntest/harness.go

```diff
@@ -163,17 +164,22 @@ func (n *NetworkHarness) SetUp(t *testing.T,
 
 	// Start the initial seeder nodes within the test network, then connect
 	// their respective RPC clients.
-	var wg sync.WaitGroup
-	wg.Add(2)
-	go func() {
-		defer wg.Done()
-		n.Alice = n.NewNode(t, "Alice", lndArgs)
-	}()
-	go func() {
-		defer wg.Done()
-		n.Bob = n.NewNode(t, "Bob", lndArgs)
-	}()
-	wg.Wait()
+	eg := errgroup.Group{}
+	eg.Go(func() error {
```

Comment:
If we already use errgroup, we can still use `n.NewNode`?  Or if we switch to `n.newNode`, it seems we don't need to use errgroup for error sync?

I think the problem came from that `testing.T` being used inside the goroutine, remembered to have issues before.

Reply:
I didn't want to touch/change the `n.NewNode()` method because that's used in 108 places.
Using `n.NewNode()` with `testing.T` is fine since we don't normally do that in goroutines. The problem here only exists because of the goroutine, that's correct. So basically any `require.xxx` within a goroutine will not reliably fail the test (or at least not immediately) so we can run into issues like that.